### PR TITLE
Improve support for Facebook permalinks

### PR DESCRIPTION
**Pull request by davisshaver**
See https://github.com/fusioneng/shortcake-bakery/issues/13

### CODE REVIEW

**Comment by davisshaver** [comment]
Trying to debug an issue with the Facebook embed rendering for logged-out users. Same markup is generated for the embed, same scripts in the HTML document.

File: inc/shortcodes/class-facebook.php [hunk 1]
```diff
@@ -79,7 +79,14 @@ public static function callback( $attrs, $content = '' ) {
...
 
-		if ( ! preg_match( '#https?://(www)?\.facebook\.com/[^/]+/posts/[\d]+#', $attrs['url'] ) && ! preg_match( '#https?://(www)?\.facebook\.com\/video\.php\?v=[\d]+#', $attrs['url'] ) && ! preg_match( '#https?:\/\/www?\.facebook\.com\/+.*?\/videos\/[\d]+\/#', $attrs['url'] ) ) {
+		// kses converts & into &amp; and we need to undo this
+		// See https://core.trac.wordpress.org/ticket/11311
+		$attrs['url'] = str_replace( '&amp;', '&', $attrs['url'] );
+
+		if ( ! preg_match( '#https?://(www)?\.facebook\.com/[^/]+/posts/[\d]+#', $attrs['url'] )
+		&& ! preg_match( '#https?://(www)?\.facebook\.com\/video\.php\?v=[\d]+#', $attrs['url'] )
+		&& ! preg_match( '#https?:\/\/www?\.facebook\.com\/+.*?\/videos\/[\d]+\/#', $attrs['url'] )
+		&& ! preg_match( '#https?://(www)?\.facebook\.com\/permalink\.php\?story_fbid=[\d]+&id=[\d]+#', $attrs['url'] ) ) {
```
Comment (danielbachhuber) [inline inc/shortcodes/class-facebook.php]: Should we put all of these patterns in an array?

Comment (davisshaver) [inline inc/shortcodes/class-facebook.php]: @danielbachhuber Yep

**Comment by davisshaver** [comment]
Here's the console log I'm seeing...
screen shot 2015-07-06 at 11 11 11 am

Unclear to me what may be causing this issue..

**Comment by davisshaver** [comment]
- Same sdk js loaded logged in and logged out
- Same markup logged in and logged out

**Comment by danielbachhuber** [comment]
Hm, using `[fusion-facebook-post url="https://www.facebook.com/permalink.php?story_fbid=615251015278265&id=380505205419515"]` seems to work for me:

image

Which URL are you attempting to embed?

**Comment by danielbachhuber** [comment]
Given there aren't any JS changes in this pull request, it seems like it might be something in your local environment.

**Comment by davisshaver** [comment]
@danielbachhuber Thanks to @goldenapples, we just diagnosed the bug as relating to Facebook permissions. Thinking about how to handle this now. It's possible we could check if the OG object is public as part of the callback process. That should probably be a new issue/PR though, as it affects existing Facebook shortcode URL's same as this new pattern. For this purpose I have found a

**Comment by danielbachhuber** [comment]
> we just diagnosed the bug as relating to Facebook permissions.

Ah, this makes sense.

> Thinking about how to handle this now. It's possible we could check if the OG object is public as part of the callback process. That should probably be a new issue/PR though, as it affects existing Facebook shortcode URL's same as this new pattern.

Let's just file a new issue for it now, and we can follow up later.

**Pull request merged**